app/jobs/google_scraping_job.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'httparty'
+require 'nokogiri'
+
+class GoogleScrapingJob < ApplicationJob
+  def perform(keyword_id, keyword)
+    scrap_from_google(keyword)
+    store_result(keyword_id)
+  end
+
+  private
+
+  def scrap_from_google(keyword)
+    uri = URI.parse("https://www.google.com/search?q=#{CGI.escape(keyword)}")
+    user_agent = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) ' \
+    'Chrome/85.0.4183.121 Safari/537.36'
+
+    response = HTTParty.get(uri, { headers: { 'User-Agent': user_agent } })
+    @parse_page = Nokogiri::HTML(response)
+  end
+
+  def store_result(keyword_id)
+    ActiveRecord::Base.transaction do
+      Keyword.where(id: keyword_id).update(
+        status: 'processed',
+        top_pos_adwords: count_top_position_adwords,
+        adwords: count_total_adwords,
+        non_adwords: count_non_adwords,
+        links: count_links,
+        html_code: @parse_page
+      )
+
+      create_top_position_adword_links(keyword_id)
+      create_non_adword_links(keyword_id)
+    end
+  end
+
+  def create_top_position_adword_links(keyword_id)
+    return if fetch_top_position_adwords_links.blank?
+
+    bulk_data = []
+
+    fetch_top_position_adwords_links.each do |link|
+      bulk_data << {
+        keyword_id: keyword_id,
+        link: link,
+        created_at: Time.current,
+        updated_at: Time.current
+      }
+    end
+
+    # rubocop:disable Rails/SkipsModelValidations
+    TopPositionAdwordLink.insert_all(bulk_data)
```

Comment:
IMO no need to create a separate table for this, we can store the `top_position_adword_links` as a column in Keyword table, same for `NonAdwordLink`

Reply:
I thought about that to but I think the column type should be `json`, so I think separate table is better.
What do you think?

Reply:
First of all, what is the use case for that column? I assume we only display the value, right? so it can be as a column?

And What's wrong with the json type, it can't store in a column 🤔 ?

Separate to another table which a different record per link, that can be a problem as the `TopPositionAdwordLink` can become really big, think about if we have 1 keyword and it has 10 links, the `TopPositionAdwordLink` will have 10 records, and if we just have 10.000 keywords, the TopPositionAdwordLink tables easily having 1.000.000 records 🤷‍♂️

Reply:
Good catch! I will make the change.

Reply:
Fixed in 8cf26122239e314ed0f18d189d8677fceac317ad ef1a28f45357ec278aa6460834987aaa5eacaed7 9d8ed7956e8a24bd5d7604c31310c0779ebef483 538ebf2211418685ea974f56ebe1482658cf3717 50a5fc1988b694c59064c437d8ba29a82a2a8161 f8daa173cd683939022b5d07ff8e9dd0e863278e 980e7f1d911981f9840d6b025cf45613543a7bb1 e8eea7d3d0472fbfc2016cf24c88b54c8592aa14